PWGHF/D2H/Tasks/taskCharmPolarisation.cxx

```diff
@@ -1840,6 +2059,152 @@ struct TaskPolarisationCharmHadrons {
   }
   PROCESS_SWITCH(TaskPolarisationCharmHadrons, processDstarMcWithMl, "Process Dstar candidates in MC with ML", false);
 
+  void processDstarInPbPb(CollsWithQvecs::iterator const& collision,
+                          FilteredCandDstarWSelFlagAndMl const& dstarCandidates,
+                          TracksWithExtra const& tracks)
+  {
+    float centrality = {-1.f};
+    centrality = o2::hf_centrality::getCentralityColl(collision, centEstimator);
+    if (centrality < minCent || centrality > maxCent) {
+      return; // skip this collision if outside of the centrality range
+    }
+    auto thisCollId = collision.globalIndex();
+    int numPvContributors = collision.numContrib();
+    auto groupedDstarCandidates = dstarCandidates.sliceBy(dstarPerCollision, thisCollId);
+    int nCands{0}, nCandsInSignalRegion{0};
+
+    std::vector<float> qVecs = getQvec(collision);
+    float xQVec = qVecs[0];
+    float yQVec = qVecs[1];
+    // float amplQVec = qVecs[2];
+    float evtPl = epHelper.GetEventPlane(xQVec, yQVec, harmonic);
```

Comment:
I would directly pass the Q-vector components to the run function (no need to compute the EP angle)
Same also below